eng/native/build-commons.sh

```diff
@@ -262,7 +262,8 @@ __msbuildonunsupportedplatform=0
 # processors available to a single process.
 platform="$(uname)"
 if [[ "$platform" == "FreeBSD" ]]; then
-  __NumProc=$(sysctl hw.ncpu | awk '{ print $2+1 }')
+  output=("$(sysctl hw.ncpu)")
+  __NumProc="$((output[1] + 1))"
```

Comment:
I think wrapping `$(sysctl hw.ncpu)` in quotes was not correct as it returns a string similar to `hw.ncpu: 4` and we need to extract the second item. With the quotes added, `output[1]` becomes an empty string and `__NumProc` will be always `1`.

This might work better:
```bash
__NumProc=$(($(sysctl -n hw.ncpu)+1))
```

Reply:
This issue is fixed by #50691.